### ISSUE TITLE
aminoacid deletions output

### ISSUE THREAD

**Issue by TYMichaelsen** [nextstrain/nextclade#274]
Thanks for a great and easy to use software! 

It looks like nextclade does not report aa deletions directly, while nextstrain+auspice does - is this deliberate? I understand that nextclade reports nt deletions as ranges, but (for me at least) it's intuitive that nextclade and nextstrain+auspice gives somewhat similar output, given it's common nextstrain origin? Reporting aa deletions directly seems like a quite important thing to do, in the context of our work its mandatory with aa deletions - so we would have to write an aditional script to fetch this information. But i'm not a virologist so there might be an obvious reason for this (in that case, sorry!).

**Comment by rneher** [comment]
Hi Thomas,

thanks for flagging this. We should definitely report amino acid deletions -- this was a "good enough for now" decision when we first implemented this. 

Currently, we have two road-blocks here. One is that the alignment we use doesn't necessarily put the gaps in frame. So alignments like
```
ATG.AGG.CAC.ACG.CGA
ATG.A--.---.-CG.CGA
```
and
```
ATG.AGG.CAC.ACG.CGA
ATG.---.---.ACG.CGA
```
are ambiguous right now. 

The other is purely an implementation issue. We currently only use substitutions to compare to the tree and only assign amino acid changes based on that. We should definitely fix that....

**Comment by TYMichaelsen** [comment]
Hi Richard,

Thanks for the reply. I see the problem. This ambiguity will be reported by nextstrain/augur using the "*" annotation right? If so that would be sufficient for now, my main concern is that we get the systematic signal and that the position is in close proximity of the actual site. I'm not too concern with if we can call it correctly - if the site turns out explanatory one would go in anyway and manually look at it - I presume? :) Please correct me if wrong. 

To summarise, from our point of view fixing "road-block two" first would get us quite far, combined with reporting ambiguous sites for "road-block one". 

Thanks!

Best,
Thomas

**Comment by rneher** [comment]
Translation in augur (from biopython) will call partial codons as `X`. The ambiguity as to where to put the gap is rarely more than one or two bases and shouldn't affect the interpretation of the gaps too much. In the case of the 69/70 deletion, one amino acid is always deleted and you get `AX-XC` as translations when it should be `AB--C` where  I am using `ABC` for other amino acids (not the actual ones, there is no `B`). 

we'll think about how to best do this. for the time being, parsing the list of gaps is what you'd need to do.

**Comment by TYMichaelsen** [comment]
Awesome! So to be sure, parsing the gaps will not be implemented in nextclade? I would have to write my own parser?

**Comment by ivan-aksamentov** [comment]
Hi Thomas @TYMichaelsen,

We've been working on the partial fix for this issue in #276 
It adds aminoacid deletions and fixes some of the problems with aminoacid substitutions.

The preliminary version of Nextclade web application with these changes is available here:
https://nextclade-git-feat-aa-deletions.nextstrain.vercel.app/

Could you give it a try and to tell if that works better for you?

You can also try to build and run the command-line version like this:

```bash
git clone https://github.com/nextstrain/nextclade -b feat/aa-deletions
yarn install
yarn cli:dev

# in a separate terminal instance
../cli/dist/nextclade.js --input-fasta=sequences.fasta --output-json=out.json --output-tsv=out.tsv
```


Note that the problem with ambiguous alignment (gap placement) which Richard mentioned above is still there. Fixing it will require implementing another alignment algorithm. We are working on it and plan to eventually port it to Nextclade, but that's a much longer adventure.

**Comment by TYMichaelsen** [comment]
Thanks a lot Ivan, I will give it a go!

**Comment by TYMichaelsen** [comment]
I tried to run the prelim version on our data, it looks quite nice and easy to understand the meaning of columns. Very useful! By comparing against the output from the online running version of nextclade the new "aaSubstitions" corresponds very well to the old "aminoacidChanges" column - also nice with reporting of ambigous aa's! 

for our purposes this is much more useful, so we will probably go with this for now. I hope it could be included to the master.

**Comment by TYMichaelsen** [comment]
Any news on merging this into the master? It seems like good to go? Sorry for pushing, we would just really like to start using it in our standard pipeline! :)

**Comment by ivan-aksamentov** [comment]
@TYMichaelsen Hi Thomas, 

Sorry for the delay. We are actually checking a few things before merging it, to make sure we don't break anything for other users.

Just to clarify, do you need it specifically on `master` branch (because you are using it in a fork for example) or do you need it to be released to https://clades.nextstrain.org? These are not the same thing, because merging to master branch does not imply release to the live website quite yet.

Or do you need a command-line NPM/Docker release? In this case I could push an "emergency" alpha release for you.

**Comment by ivan-aksamentov** [comment]
@TYMichaelsen Hi Thomas, 

Sorry for the delay. We are actually checking a few things before merging it, to make sure we don't break anything for other users.

Just to clarify, do you need it specifically on `master` branch (because you are using it in a fork for example) or do you need it to be released to https://clades.nextstrain.org? These are not the same thing, because merging to master branch does not imply release to the live website quite yet.

Or do you need a command-line NPM/Docker release? In this case I could push an "emergency" alpha release for you.

**Comment by TYMichaelsen** [comment]
Hi Ivan,
No problem, I figured there must be a reason for the delay :) 

We use the command-line version and the reason I'm asking is basically just to make sure that the things you have implemented are good-to-go. In that case an alpha-version sounds perfect!

**Comment by ivan-aksamentov** [comment]
Released in version 0.10.0
